docs/modules/ROOT/pages/building.adoc

```diff
@@ -0,0 +1,91 @@
+= Building the Operator
+
+This operator is written in Rust.
+
+It is developed against the latest stable Rust release, and we currently don't support any older versions.
+
+The Listener Operator is a https://github.com/container-storage-interface/spec/blob/master/spec.md[Container Storage Interface (CSI)] provider plugin
+for the local Kubelet, which means that it should only be executed inside a Kubernetes `Pod`. Currently, the following ways are supported to build the Listener Operator:
+
+* `docker build` 
+* https://nixos.org/[Nix]
+
+The `docker build`command is currently the primary deployment target, and the official images are built
+using it. However, Nix has much faster incremental build and deploy times, making it ideal for local development.
+
+== Docker
+
+To build and deploy to the active Kind cluster, run:
+
+[source,console]
+----
+$ echo Building with Docker
+# Ensure that all submodules are up-to-date
+$ git submodule update --recursive --init
+# Update the Chart metadata and CRD definitions
+$ make compile-chart
+# Create a unique image ID
+$ REPO=listener-operator
+$ TAG="$(uuidgen)"
+# Build the image
+$ docker build . -f docker/Dockerfile -t "$REPO:$TAG"
+# Load the image onto the Kind nodes
+$ kind load docker-image "$REPO:$TAG"
+# Deploy
+$ helm upgrade listener-operator deploy/helm/listener-operator \
+       --install \
+       --set-string "image.repository=$REPO,image.tag=$TAG"
+----
+
+== Nix
+
+To build and deploy to the active Kind cluster, run the following bash script:
+
+[source,console]
+----
+$ echo Building with Nix
+# Ensure that all submodules are up-to-date
+$ git submodule update --recursive --init
+# Use crate2nix (https://github.com/kolloch/crate2nix) to convert Cargo.lock into a Nix derivation
+$ nix run -f . crate2nix generate
+# Build the Docker images
+# A custom Docker repository can be specified by appending `--argstr dockerName my-custom-repository`
+$ nix build -f . docker
+# Load the images onto the Kind nodes
+# Nix does not use the Docker daemon, instead it builds individual layers, as well as
+# a script (`result/load-image`) that combines them into a Docker image archive.
+# `load-image` can also be piped into `docker load` to prepare for pushing to a registry.
+# Some people have reported issues with piping, for those it may be helpful to buffer the image through a file
+# by running `./result/load-image > image.tar && kind load image-archive image.tar`.
+$ ./result/load-image | kind load image-archive /dev/stdin
```

Comment:
I get the same error as Sigi here: https://github.com/stackabletech/secret-operator/pull/181#discussion_r967018758

Reply:
I think I finally understand what's going on, please check if1b0e70b382e2d7ec0653e82f7ee7cb1df65b4d6e seems to match your experience.